src/quantile/mod.rs

```diff
@@ -381,19 +211,81 @@ where
     /// - worst case: O(`m`^2);
     /// where `m` is the number of elements in the array.
     ///
-    /// **Panics** if `axis` is out of bounds, if the axis has length 0, or if
+    /// Returns `Err(EmptyInput)` when the specified axis has length 0.
+    ///
+    /// **Panics** if `axis` is out of bounds or if
     /// `q` is not between `0.` and `1.` (inclusive).
```

Comment:
Since we're returning a `Result` now and we iterate over `qs` to check this constraint anyway, we might as well make invalid `q` values result in an error instead of panic. We could add another error type:

```rust
/// An error computing a quantile.
pub enum QuantileError {
    /// The input was empty.
    EmptyInput,
    /// The `q` was not between `0.` and `1.` (inclusive).
    InvalidFraction(N64),
}
```

I'm not really sure what to call that second variant. `InvalidFraction` is okay, but I'd be happy for other suggestions.